py-polars/polars/internals/lazyframe/frame.py

```diff
@@ -2566,7 +2566,7 @@ def with_columns(
             exprs = list(exprs)
 
         exprs.extend(
-            pli.expr_to_lit_or_expr(expr).alias(name)
+            pli.expr_to_lit_or_expr(expr, structify=True).alias(name)
```

Comment:
I think `structify` should only be `True` if `named_exprs` are given. The meta processing involved is relatively expensive and unneeded in many cases.

Reply:
This Is already functionally the case ;) Nothing will execute in that loop if there aren't any named_exprs, due to `for name, expr in named_exprs.items()`.

Reply:
Oh, right. List comprehension. So the for is after the expression. Now I see it. :+1: